src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/SearchADTasksAction.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.ad.transport;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.search.SearchResponse;
+
+import com.amazon.opendistroforelasticsearch.ad.constant.CommonValue;
+
+public class SearchADTasksAction extends ActionType<SearchResponse> {
```

Comment:
What's the use case to call this api?

Reply:
For current phase, this is only used for Ops purpose. User can use this to search historical tasks as we don't show task level data on Kibana. 
From UX, we may develop task management page on Kibana in future, then we can use this API to search tasks and show them on Kibana.